features/dd-sdk-android-flags/src/test/kotlin/com/datadog/android/flags/utils/forge/PrecomputedFlagForgeryFactory.kt

```diff
@@ -22,34 +22,30 @@ internal class PrecomputedFlagForgeryFactory : ForgeryFactory<PrecomputedFlag> {
             VariationType.BOOLEAN -> VariationType.BOOLEAN.value
             VariationType.STRING -> VariationType.STRING.value
             VariationType.INTEGER -> VariationType.INTEGER.value
-            VariationType.DOUBLE -> VariationType.DOUBLE.value
-            VariationType.JSON -> VariationType.JSON.value
+            VariationType.NUMBER -> VariationType.NUMBER.value
+            VariationType.FLOAT -> VariationType.FLOAT.value
+            VariationType.OBJECT -> VariationType.OBJECT.value
         }
 
         val variationValue = when (variationTypeEnum) {
             VariationType.BOOLEAN -> forge.aBool().toString()
             VariationType.STRING -> forge.anAlphabeticalString()
             VariationType.INTEGER -> forge.anInt().toString()
-            VariationType.DOUBLE -> forge.aDouble().toString()
-            VariationType.JSON -> JSONObject().put(
+            VariationType.NUMBER -> forge.aDouble().toString()
+            VariationType.FLOAT -> forge.aFloat().toString()
+            VariationType.OBJECT -> JSONObject().put(
                 "key",
                 forge.anAlphabeticalString()
             ).toString()
         }
 
-        val reasonEnum = forge.anElementFrom(*FlagReason.values())
-        val reason = when (reasonEnum) {
-            FlagReason.DEFAULT -> FlagReason.DEFAULT.value
-            FlagReason.TARGETING_MATCH -> FlagReason.TARGETING_MATCH.value
-            FlagReason.RULE_MATCH -> FlagReason.RULE_MATCH.value
-            FlagReason.PREREQUISITE_FAILED -> FlagReason.PREREQUISITE_FAILED.value
-            FlagReason.ERROR -> FlagReason.ERROR.value
-        }
+        val reasonEnum = forge.aValueFrom(ResolutionReason::class.java)
+        val reason = reasonEnum.name
 
         return PrecomputedFlag(
             variationType = variationType,
             variationValue = variationValue,
-            doLog = forge.aBool(),
+            doLog = true, // Default to true - tests can override if needed
```

Comment:
minor: I think we should consider forging a boolean value and overriding in tests as necessary, otherwise this forgery factory is misleading because it isn't actually forging all cases and this could be missed later on.